### PR TITLE
Initial Digits Setup

### PR DESCRIPTION
**Pull request by araines**

### CODE REVIEW

File: src/Client.php [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+use Sportlobster\Digits\Exception\KeyMismatchException;
+use Symfony\Component\Serializer\Encoder\JsonEncoder;
+use Symfony\Component\Serializer\NameConverter\CamelCaseToSnakeCaseNameConverter;
+use Symfony\Component\Serializer\Normalizer\GetSetMethodNormalizer;
+use Symfony\Component\Serializer\Serializer;
+use Symfony\Component\Serializer\SerializerInterface;
+
+class Client
+{
+    const HOST_DIGITS = 'api.digits.com';
```
Comment (konybakk) [inline src/Client.php:21]: I think this is going to make it very hard to functionally test the app end-to-end using behat and wiremock as you can't change the host to hit wiremock. I see why you've done it though, the host is used for security, so not sure what to do about it.

File: spec/ClientSpec.php [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        $serializer->deserialize($body, User::class, 'json')->willReturn($user);
+
+        $this->beConstructedWith('SomeKey');
+        $this->setHttpClient($httpClient);
+        $this->setSerializer($serializer);
+        $this->verify($url, $auth)->shouldReturn($user);
+    }
+
+    public function it_throws_an_authentication_exception_if_could_not_verify_user(
+        ClientInterface $httpClient, SerializerInterface $serializer,
```
Comment (konybakk) [inline spec/ClientSpec.php]: only httpClient is needed, the others aren't used.

**Review by konybakk — CHANGES_REQUESTED** [review]

File: src/Client.php (hunk 1 above)
Comment (araines) [reply src/Client.php:21]: Plan for testing was to configure up an http client (Guzzle) which has a proxy in it, allowing for the traffic to be intercepted by a test mock (WireMock).

Comment (konybakk) [reply src/Client.php:21]: but you can't do that from the api ... so you have no way of doing a black box test from the api...

**Review by konybakk — APPROVED** [review]

**Pull request merged**